modules/stochastic_tools/test/tests/multiapps/batch_commandline_control/master_vector.i

```diff
@@ -0,0 +1,104 @@
+[StochasticTools]
+[]
+
+[Distributions]
+  [uniform]
+    type = Uniform
+    lower_bound = 5
+    upper_bound = 10
+  []
+[]
+
+[Samplers]
+  [sample]
+    type = MonteCarlo
+    num_rows = 3
+    distributions = 'uniform uniform uniform uniform'
+    execute_on = 'PRE_MULTIAPP_SETUP'
+  []
+[]
+
+[MultiApps]
+  [sub]
+    type = SamplerFullSolveMultiApp
+    sampler = sample
+    input_files = 'sub.i'
+    mode = batch-reset
+  []
+[]
+
+[Transfers]
+  [data]
+    type = SamplerPostprocessorTransfer
+    multi_app = sub
+    sampler = sample
+    to_vector_postprocessor = storage
+    from_postprocessor = size
+  []
+  [prop_A]
+    type = SamplerPostprocessorTransfer
+    multi_app = sub
+    sampler = sample
+    to_vector_postprocessor = prop_A
+    from_postprocessor = prop_A
+  []
+  [prop_B]
+    type = SamplerPostprocessorTransfer
+    multi_app = sub
+    sampler = sample
+    to_vector_postprocessor = prop_B
+    from_postprocessor = prop_B
+  []
+  [prop_C]
+    type = SamplerPostprocessorTransfer
+    multi_app = sub
+    sampler = sample
+    to_vector_postprocessor = prop_C
+    from_postprocessor = prop_C
+  []
+  [prop_D]
+    type = SamplerPostprocessorTransfer
+    multi_app = sub
+    sampler = sample
+    to_vector_postprocessor = prop_D
+    from_postprocessor = prop_D
+  []
+[]
+
+[VectorPostprocessors]
+  [storage]
+    type = StochasticResults
+  []
+  [prop_A]
+    type = StochasticResults
+  []
+  [prop_B]
+    type = StochasticResults
+  []
+  [prop_C]
+    type = StochasticResults
+  []
+  [prop_D]
+    type = StochasticResults
+  []
+  [sample_data]
+    type = SamplerData
+    sampler = sample
+  []
+[]
+
+[Controls]
+  [cmdline]
+    type = MultiAppCommandLineControl
+    multi_app = sub
+    sampler = sample
+    param_names = 'Mesh/xmax[0] Materials/const/prop_values[1,(1.5),2,2] Mesh/ymax'
```

Comment:
I am thinking that if you have [] then they all need them?

Reply:
Current implementation will set ymax as the ymax[3].

I like your idea that if you want to use [], then you have to provide [] for all. I will make the change. I guess for vector parameter, we need [] anyway because we do not know the size of the vector through command line. Are you Ok with that? Or do you have a better way that does not need [] for vector parameter?

Reply:
I am OK with requiring it for a vector, since we don't know the size there really isn't a better option.